### PR TITLE
refactor: [M3-6369] - MUI v5 - `Components > PaginationControls, PaginationFooter`

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Style migration for `<PaginationControls />` and  `<PaginationFooter />`

## Major Changes 🔄
- Use MUI's `<Pagination />` instead of our custom component
- Fixes the weird spacing that has existed for years with this component

### Before 😵‍💫
Screenshot 2023-06-09 at 11 27 43 AM

### After 🎉
Screenshot 2023-06-09 at 11 27 22 AM

## How to test 🧪
- Test pagination throughout the app
  - You may need to use the MSW to test this if you don't have any large tables
  - A good place to test would be Account Maintenance with the MSW on
    - You can edit line 756 of `serverHandlers.ts` to make lots of account maintenance rows

### CODE REVIEW

**Review by mjac0bs — COMMENTED** [review]
This component switch is looking good! What do you think about adjusting the background color for the selected page number in dark mode? Right now it's so dark that there's not much contrast, which could make it difficult for the user to see which page they are currently on.

This branch:
Screenshot 2023-06-12 at 3 01 09 PM

`develop`:
Screenshot 2023-06-12 at 3 02 02 PM

**Review by hana-linode — COMMENTED** [review]
Nice cleanup!

The dark mode colors seem a bit off. The hover and selected state contrast isn't great and the white text on a dark background is harsh

Prod

https://github.com/linode/manager/assets/115299789/560c82ca-9953-476e-85e9-3221f989fbb7




This branch


https://github.com/linode/manager/assets/115299789/3b7bdf63-73a1-4c79-a94b-67a208e0b603

**Comment by bnussman-akamai** [comment]
Thank you for the feedback @mjac0bs and @hana-linode. I was able to fix the contrast issues by fixing an even larger underlying problem 53188d9 🎉

**Review by jaalah-akamai — APPROVED** [review]

File: packages/manager/src/components/PaginationControls/PaginationControls.tsx [hunk 1]
```diff
@@ -1,136 +1,22 @@
...
-const calNumOfPages = (count: number, pageSize: number) =>
-  Math.ceil(count / pageSize);
+export const PaginationControls = (props: Props) => {
+  const { count, page, pageSize, onClickHandler } = props;
+
+  return (
+    <Pagination
+      count={Math.ceil(count / pageSize)}
+      page={page}
+      onChange={(event, page) => onClickHandler(page)}
```
Comment (abailly-akamai) [inline packages/manager/src/components/PaginationControls/PaginationControls.tsx:18]: You can get rid of `event` here

File: packages/manager/src/foundations/themes/dark.ts [hunk 2]
```diff
@@ -155,8 +155,7 @@ export const darkTheme: ThemeOptions = {
     ...iconCircleAnimation,
   },
   palette: {
-    // We really should be using this...
-    // mode: 'dark',
+    mode: 'dark',
```
Comment (abailly-akamai) [inline packages/manager/src/foundations/themes/dark.ts:158]: can you explain the reason for re-introducing this?

File: packages/manager/src/foundations/themes/light.ts [hunk 3]
```diff
@@ -166,7 +166,7 @@ const graphTransparency = '0.7';
 const spacing = 8;
 
 export const lightTheme: ThemeOptions = {
-  name: 'light', // we really should just leverage pallete.mode
+  name: 'light', // @todo remove this because we leverage pallete.mode now
```
Comment (abailly-akamai) [inline packages/manager/src/foundations/themes/light.ts:169]: same. why re-introducing this?

**Review by abailly-akamai — COMMENTED** [review]
Looks good! a few improvements and questions

File: packages/manager/src/foundations/themes/light.ts (hunk 3 above)
Comment (bnussman-akamai) [reply packages/manager/src/foundations/themes/light.ts:169]: We aren't reintroducing. This is something I discovered when I did the MUI v4 to v5 migration and left a comment.

Here I'm suggesting we delete `name` from our MUI theme (which requires module augmentation) in favor of their built in way of doing it, which is using `pallete.theme`.

File: packages/manager/src/foundations/themes/light.ts [hunk 4]
```diff
@@ -284,6 +284,7 @@ export const lightTheme: ThemeOptions = {
     ...genericTableHeaderStyle,
   },
   palette: {
+    mode: 'light',
```
[the remark this answers is not in the document]
Comment (bnussman-akamai) [reply packages/manager/src/foundations/themes/light.ts:287]: Using `palette.mode` is MUI's built in way of distinguishing between light and dark theme. To me this is much cleaner than relying on our custom way of naming the theme. 

This also causes MUI to switch some colors out in dark mode which addresses @mjac0bs and @hana-linode's feedback without me needing to really do anything.

File: packages/manager/src/components/PaginationControls/PaginationControls.tsx (hunk 1 above)
Comment (bnussman-akamai) [reply packages/manager/src/components/PaginationControls/PaginationControls.tsx:18]: Replace it with a `_`?

File: packages/manager/src/foundations/themes/light.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (abailly-akamai) [reply packages/manager/src/foundations/themes/light.ts:287]: gotcha thanks for the extra context

**Review by mjac0bs — APPROVED** [review]

**Pull request merged**